tad/services/projects.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from typing import Annotated
+
+from fastapi import Depends
+
+from tad.core.config import get_settings
+from tad.models import Project
+from tad.repositories.projects import ProjectsRepository
+from tad.schema.instrument import InstrumentBase
+from tad.schema.project import ProjectNew
+from tad.schema.system_card import SystemCard
+from tad.services.storage import WriterFactory
+
+logger = logging.getLogger(__name__)
+
+
+class ProjectsService:
+    def __init__(self, repository: Annotated[ProjectsRepository, Depends(ProjectsRepository)]) -> None:
+        self.repository = repository
+
+    def get(self, project_id: int) -> Project | None:
+        project = None
+        try:
+            project = self.repository.find_by_id(project_id)
+        except Exception:
+            project = None
+
+        return project
+
+    def create(self, project_new: ProjectNew) -> Project:
+        project = Project(name=project_new.name)
+
+        self.repository.save(project)
+
+        system_card_file = get_settings().CARD_DIR / f"{project.id}_system.yaml"
+
+        project.model_card = str(system_card_file)
+
+        project = self.update(project)
+
+        instruments: list[InstrumentBase] = [
```

Comment:
(re-commenting because I think it got lost due to refactoring this code):
When I select 1 instrument, it loops over the string instead of the list, and thus creates a system card containing instruments of 1 letter (being all the letters of the chosen instrument urn) instead of 1 instrument with the full urn.

Reply:
I will have a look. indeed missed this one

Reply:
@uittenbroekrobbert . Nice find. solved it in the schema validator